### DISCUSSION TITLE
Catalog consumer failsafe mode

### DISCUSSION THREAD

**Discussion by klaus-nicat** [PowerDNS/pdns#14762] [Ideas]
Hi all! I am thinking about catalog zones for our customers - they are the producer and we are the consumer. But I am concerned, that if the customer does something the wrong way, all his zones will be deleted immediately and all the customer zones are offline.

For example: the customer has an outage on his hidden primary, and then he re-setups his hidden primary. When he, by mistake, first configures the catalog producer (but no member zones yet), the catalog AXFR will provide us with an empty catalog and our PowerDNS immediately deletes all member zones of this customer. Or to cite RFC9432: "Great power comes with great responsibility."

Honestly, this is the only reason why we do not yet offer catalog zones to customers but only use them internally.

So I wonder if we can add some safeguard feature (configureable), eg. like `catalog-reject-empty=yes|no` or `catalog-min-members=<membercount>`, so when the catalog is empty or the member count is below a certain threshold, then PowerDNS logs an error but does not consume the catalog (for real cleanup I would implement an out-of-catalog-zone mechanism). Maybe the threshold can also be configured (customers with 1mio zones would use different thresholds than customers with 1000 zones).

**Comment by klaus-nicat** [comment]
I just stumbled over #15093 which sounds like a proof that my above suggestions would make sense to detect upstream problems and refuse to work with emtpy catalogs. Still not interested in my ideas, even if we provide patches?

**Comment by omoerbeek** [comment]
Although I'm not the auth maintainer I think it's a useful safety feature and we would be interested in a PR.
